Fix messaging

ptgott opened the pull request:
In gravitational/docs#238, we will add a linter that lints for incorrect usage of Teleport terms. This change fixes linter violations to anticipate this new linter.

zmb3 commented on docs/pages/access-controls/guides/hardware-key-support.mdx:
```diff
@@ -27,22 +27,22 @@ Additionally, this feature can be configured to require touch for every Teleport
     - Standard Teleport API requests (`tsh ls`, `tctl create`, etc.)
-    - Server Access
-    - Database Access
+    - Teleport SSH Service 
```
This doesn't seem right to me. This docs page is attempting to say that hardware key support works for accessing SSH servers or databases. Why are we mentioning the specific service in the Teleport process here?

zmb3 commented on docs/pages/architecture/session-recording.mdx:
```diff
@@ -49,9 +49,9 @@ Windows, database, and Kubernetes sessions are always recorded at the host runni
-`proxy-sync` identically (perform synchronous recording).
+For this reason, the Teleport Desktop, Database, and Kubernetes Services treat
+`node` and `proxy` identically (perform asynchronous recording) and `node-sync`
+and `proxy-sync` identically (perform synchronous recording).  
```
Remove trailing whitespace.

zmb3 commented on docs/pages/connect-your-client/introduction.mdx:
```diff
@@ -310,8 +310,8 @@ either directly or through proxy tunnels.
 - Teleport Connect is a graphic utility for connecting to resources through
-Teleport. You can initiate Server Access, connect to databases, and start
-Desktop Access sessions. See [Using Teleport Connect](./teleport-connect.mdx).
+  Teleport. You can connect to servers, databases, and desktops. See [Using
```
I don't know how this got here, but it's incorrect. You can not connect to desktops with Teleport Connect - only servers, databases, and Kube clusters.

zmb3 commented on docs/pages/database-access/architecture.mdx:
```diff
@@ -1,13 +1,14 @@
-title: Database Access Architecture
-description: How Teleport Database Access works.
+title: Database Service Architecture
+description: How Teleport the Database Service works.
```
This page isn't describing the architecture of just the database service, it's describing the architecture of the entire database access feature.

If you don't want to say _Database Access_ anymore, that's fine, but I think we should find a title that makes it clear this  is about how Teleport provides access to databases, which involves not only the database service, but the auth service and proxy service.

zmb3 commented on docs/pages/desktop-access/active-directory-manual.mdx:
```diff
@@ -1,25 +1,25 @@
 ---
-title: Desktop Access with Active Directory (Manual)
+title: Register Windows Desktops with Active Directory (Manual)
```
This guide isn't really about registering desktops. How about "Connecting to desktops" instead?

zmb3 commented on docs/pages/preview/upcoming-releases.mdx:
```diff
@@ -22,13 +22,13 @@ You can find the full list of fixes and features in the [GitHub milestone](https
-session recording, per-session MFA, and clipboard support. Database Access
-will add self-hosted Redis, auto-discovery for Redshift clusers, and
-auto-IAM configuration improvements.
+Teleport 9.0 will focus on bringing the Desktop Service to GA with support for
```
This change probably isn't necessary. Nobody should be going to the upcoming releases page anymore to find out whats in a release from 9 months ago.

zmb3 reviewed (commented):
Mostly looks fine, but I can't help but feel like we've made the docs less clear in places because we're trying so hard not to say "Server Access"

If this is the future direction, do we plan to update the sidebar entries too? Seems odd if they still say things like "Database Access" after we've removed all other references to the term

ptgott commented:
@zmb3 Thanks for the review! I'm going to tweak the linter so we don't sacrifice clarity. I think we can:

- Ignore capitalized instances of "Database Access", "Server Access", etc. in titles. Since titles capitalize significant terms anyway, readers aren't going to think these refer to separate products.
- Catch capitalized instances of "Database Access", "Server Access", etc. in the body text and page description while recommending using the lowercased version instead, e.g., "database access". This way we don't need to go through odd contortions to talk about accessing resources.

What do you think?

ptgott committed:
Tweak messaging to anticipate a new linter

In gravitational/docs#238, we will add a linter that lints for incorrect
usage of Teleport terms. This change fixes linter violations to
anticipate this new linter.

Changes include:

- Ensuring that Teleport service names are capitalized
- No longer using the "[Resource] Access" terminology. Instead, talk
  about adding resources, using Teleport services, etc.

ptgott commented:
@alexfornuto Checking to see if you have any thoughts on this PR. I think it would be great to be able to merge the [linter PR](https://github.com/gravitational/docs/pull/238) in `gravitational/docs` so we can standardize Teleport terms across the docs.
